stl/inc/xstring

```diff
@@ -2463,23 +2512,94 @@ private:
             int>;
 #endif // _HAS_CXX17
 
-public:
-    _CONSTEXPR20 basic_string(const basic_string& _Right)
-        : _Mypair(_One_then_variadic_args_t{}, _Alty_traits::select_on_container_copy_construction(_Right._Getal())) {
-        auto&& _Alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Getal());
-        _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _Mypair._Myval2);
-        _Construct_lv_contents(_Right);
-        _Proxy._Release();
+#ifdef _INSERT_STRING_ANNOTATION
+    _CONSTEXPR20 void _Create_annotation() const noexcept {
+        // Annotates the valid range with shadow memory
+        auto& _My_data = _Mypair._Myval2;
+        _Apply_annotation(_My_data._Myptr(), _My_data._Myres, _My_data._Myres, _My_data._Mysize);
     }
 
-    _CONSTEXPR20 basic_string(const basic_string& _Right, const _Alloc& _Al)
-        : _Mypair(_One_then_variadic_args_t{}, _Al) {
-        auto&& _Alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Getal());
-        _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _Mypair._Myval2);
-        _Construct_lv_contents(_Right);
-        _Proxy._Release();
+    _CONSTEXPR20 void _Remove_annotation() const noexcept {
+        // Removes annotation of the range with shadow memory
+        auto& _My_data = _Mypair._Myval2;
+        _Apply_annotation(_My_data._Myptr(), _My_data._Myres, _My_data._Mysize, _My_data._Myres);
+    }
+
+    _CONSTEXPR20 void _Modify_annotation(const difference_type _Count) const noexcept {
+        // Extends/shrinks the annotated range by _Count

```

Comment:
Remember to add test for `0` here.

Reply:
great catch, it totally nodded ahead with the vector changes and never thought of that one here